cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java

```diff
@@ -0,0 +1,53 @@
+package org.cloudfoundry.client.lib.adapters;
+
+import java.net.URL;
+
+import org.cloudfoundry.client.CloudFoundryClient;
+import org.cloudfoundry.client.lib.domain.CloudSpace;
+import org.cloudfoundry.client.lib.oauth2.OAuthClient;
+import org.cloudfoundry.operations.CloudFoundryOperations;
+import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
+import org.cloudfoundry.reactor.ConnectionContext;
+import org.cloudfoundry.reactor.DefaultConnectionContext;
+import org.cloudfoundry.reactor.TokenProvider;
+import org.cloudfoundry.reactor.client.ReactorCloudFoundryClient;
+
+public class CloudControllerV3ClientFactory {
+
+    public CloudFoundryOperations createOperationsClient(URL controllerUrl, OAuthClient oAuthClient, CloudSpace target) {
+        DefaultCloudFoundryOperations.Builder builder = DefaultCloudFoundryOperations.builder()
+            .cloudFoundryClient(createClient(controllerUrl, oAuthClient));
+        if (target != null) {
+            builder.organization(getOrganizationName(target))
+                .space(getSpaceName(target));
+        }
+        return builder.build();
+    }
+
+    private String getOrganizationName(CloudSpace target) {
+        return target.getOrganization()
```

Comment:
Above there is a check for the target whether it is null. Is this check needed here?

Reply:
Fixed.